backend/src/auth/utils/ttl.util.ts

```diff
@@ -0,0 +1,23 @@
+import { Injectable } from '@nestjs/common';
+import { AppConfigService } from '../../config/config.service';
+
+@Injectable()
+export class TtlUtil {
+  constructor(private readonly appConfigService: AppConfigService) {}
+
+  convertExpirationTimeToTtl(expirationTime: string): number {
+    if (!expirationTime) {
+      return null;
+    }
+    const expirationTimeAsNumber = new Date(expirationTime).getTime();
+    const now = new Date().getTime();
+    if (expirationTimeAsNumber < now) {
+      return null;
+    }
+    return (expirationTimeAsNumber - now) / 1000;
+  }
+
+  getSmallerTtl(ttl: number) {
+    return Math.min(ttl, this.appConfigService.getConfig().redis.ttl);
+  }
```

Comment:
I think we should consider remove this method and use it in the service itself (shortener service).
We can create a util class with exported functions. (remove Injectable from this class).